### ISSUE TITLE
Rebuilding bundles that use crypto module causes runtime errors

### ISSUE THREAD

**Issue by AgentME** [substack/watchify#130]
If you try to use watchify to rebuild a bundle that uses the crypto module, then you get a nonsense exception at runtime when you run the bundle.

See https://github.com/AgentME/br-test for a quick working(failing) example.

**Comment by kumavis** [comment]
Error is as follows:
```
/Users/kumavis/Development/Node/br-test/out.js:5148
elliptic.version = require('../package.json').version;
                                             ^
TypeError: Cannot read property 'version' of undefined
    at Object./Users/kumavis/Development/Node/br-test/node_modules/browserify/node_modules/crypto-browserify/node_modules/browserify-sign/node_modules/elliptic/lib/elliptic.js.../package.json (/Users/kumavis/Development/Node/br-test/out.js:5148:46)
    at Object./Users/kumavis/Development/Node/br-test/node_modules/browserify/node_modules/crypto-browserify/node_modules/create-ecdh/node_modules/elliptic/lib/elliptic.js.../package.json (/Users/kumavis/Development/Node/br-test/out.js:12137:182)
    at s (/Users/kumavis/Development/Node/br-test/out.js:1:316)
    at /Users/kumavis/Development/Node/br-test/out.js:1:367
    at Object.<anonymous> (/Users/kumavis/Development/Node/br-test/out.js:12012:16)
    at Object./Users/kumavis/Development/Node/br-test/node_modules/browserify/node_modules/crypto-browserify/node_modules/create-ecdh/browser.js.bn.js (/Users/kumavis/Development/Node/br-test/out.js:12128:4)
    at s (/Users/kumavis/Development/Node/br-test/out.js:1:316)
    at /Users/kumavis/Development/Node/br-test/out.js:1:367
    at Object./Users/kumavis/Development/Node/br-test/node_modules/browserify/node_modules/crypto-browserify/node_modules/create-ecdh/index.js../browser (/Users/kumavis/Development/Node/br-test/out.js:12133:32)
    at s (/Users/kumavis/Development/Node/br-test/out.js:1:316)
```

**Comment by jscissr** [comment]
Having the same issue with another package.
This seems related to #18, but that is already closed; so the issue somehow reappeared.

The problem is with JSON files. When you browserify a .json file, `module.export=` is prefixed:

```
"C:\\Users\\Jan\\Documents\\GitHub\\br-test\\node_modules\\browserify\\node_modules\\crypto-browserify\\node_modules\\browserify-sign\\node_modules\\elliptic\\package.json":[function(require,module,exports){
module.exports={
  "name": "elliptic",
```

However, on the second build this becomes:

```
"C:\\Users\\Jan\\Documents\\GitHub\\br-test\\node_modules\\browserify\\node_modules\\crypto-browserify\\node_modules\\browserify-sign\\node_modules\\elliptic\\package.json":[function(require,module,exports){
module.exports=module.exports={
  "name": "elliptic",
```

And so on. This is not yet causing any issues, but if you have the same package twice, browserify saves space by only making aliases, like this:

```
"C:\\Users\\Jan\\Documents\\GitHub\\br-test\\node_modules\\browserify\\node_modules\\crypto-browserify\\node_modules\\create-ecdh\\node_modules\\elliptic\\package.json":[function(require,module,exports){
arguments[4]["C:\\Users\\Jan\\Documents\\GitHub\\br-test\\node_modules\\browserify\\node_modules\\crypto-browserify\\node_modules\\browserify-sign\\node_modules\\elliptic\\package.json"][0].apply(exports,arguments)
},{}],
```

What now happens in the second build, is this:

```
"C:\\Users\\Jan\\Documents\\GitHub\\br-test\\node_modules\\browserify\\node_modules\\crypto-browserify\\node_modules\\create-ecdh\\node_modules\\elliptic\\package.json":[function(require,module,exports){
module.exports=arguments[4]["C:\\Users\\Jan\\Documents\\GitHub\\br-test\\node_modules\\browserify\\node_modules\\crypto-browserify\\node_modules\\browserify-sign\\node_modules\\elliptic\\package.json"][0].apply(exports,arguments)
},{}],
```

And since the `arguments[4].....apply(exports,arguments)` call only assigns the JSON object to `module.exports` and returns `undefined`, `module.exports` is now `undefined`.

**Comment by kumavis** [comment]
yikes!

**Comment by Pleochism** [comment]
It seems that including a JSON file directly is still not double-appending (which was what was fixed in #18), but referencing JSON via require in JS, does.

The problem is here: https://github.com/substack/watchify/blob/956799359ef98c80bf60ef545373663365db831d/index.js#L8

I'm not sure what interaction this has with browserify, but making the caches have different names so as not to (presumably) be changed/mangled by browserify, solves it. Tests all continue to pass. However, I assume this has a performance implication. I'm unable to determine whether this would be a safe change to PR, as I'm not familiar enough with node-browserify.

Also note that this bug has existed since commit 3358625

**Comment by kumavis** [comment]
@Pleochism can you make a PR with the changes?

**Comment by Pleochism** [comment]
Sure.

https://github.com/substack/watchify/pull/152

**Comment by zertosh** [comment]
Fixed in 2.5.0
